### PR TITLE
feat(api): notify comms for appellant case incomplete

### PR DESCRIPTION
**Pull request by BiggsKC**
## Describe your changes

- `appealIncomplete` template ID added to schema
- notifyClient sends email when appellant case is incomplete
- incomplete reasons and appeal due date formatted for template

## Issue ticket number and link

[BOAT-875 API - Comms - Appellant case marked as ‘incomplete’](https://pins-ds.atlassian.net/browse/BOAT-875)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js [hunk 1]
```diff
@@ -94,6 +95,46 @@ const updateAppellantCaseValidationOutcome = async (
...
+			throw new Error(ERROR_NO_RECIPIENT_EMAIL);
+		}
+
+		const newIncompleteReasons =
+			updatedAppeal.appellantCase?.appellantCaseIncompleteReasonsOnAppellantCases;
+		if (!newIncompleteReasons) {
+			throw new Error(ERROR_NOT_FOUND);
+		}
+
+		const incompleteReasonsList = newIncompleteReasons.flatMap((item) => {
```
Comment (glpins) [inline appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:112]: Could we consider abstracting the logic building incompleteReasonsList into a reusable function?

**Review by glpins — APPROVED** [review]

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js (hunk 1 above)
Comment (fabfab-kc) [reply appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:112]: Yes this definitely be fun to convert into email nested bullet points

Comment (fabfab-kc) [reply appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:112]: and this is a brilliant solution, of flattening them and wrap the text reasons with the parent...

**Review by fabfab-kc — APPROVED** [review]
LGTM!
As suggested, it would make sense to extract a converter for incomplete / invalid reasons to handle the bullet point conversion in the notify templates... But I think you can add a separate improvement ticket for later, or before the next email that needs reasons to be shown...

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js (hunk 1 above)
Comment (BiggsKC) [reply appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:112]: I'm thinking to refactor it when I do the invalid ticket, as that's where the reuse becomes relevant. What you think?

**Review by fabfab-kc — APPROVED** [review]
LGTM!

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js (hunk 1 above)
Comment (BiggsKC) [reply appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:112]: > and this is a brilliant solution, of flattening them and wrap the text reasons with the parent...

I will unashamedly admit I had chatGPT assist me on that one ;)

**Review by fabfab-kc — APPROVED** [review]
LGTM!

**Pull request merged**